src/index.ts

```diff
@@ -34,432 +35,409 @@ export const startVercel = async (req: VercelRequest, res: VercelResponse) => {
+  'answering-second-order.80': '80%',
+  'answering-second-order.90': '90%',
+  'answering-second-order.100': '100%',
+};
+const secondButtons = Object.keys(secondButtonOptions).map((key) =>
+  Markup.button.callback(secondButtonOptions[key], key),
+);
+const formattedSecondButtons = [
+  secondButtons.slice(0, 5), // Elements 0-4
+  secondButtons.slice(5, 6), // Element 5
+  secondButtons.slice(6), // Elements 6 and beyond
+];
+
+bot.action('answering-first-order.1', async (ctx) => {
+  ctx.reply(secondPrompt, Markup.inlineKeyboard(formattedSecondButtons));
+});
+
+bot.action('answering-first-order.2', async (ctx) => {
+  ctx.reply(secondPrompt, Markup.inlineKeyboard(formattedSecondButtons));
+});
 
-*Please respond with your email address to continue\\.*`
-  ctx.replyWithMarkdownV2(prompt)
+bot.action('answering-first-order.3', async (ctx) => {
+  ctx.reply(secondPrompt, Markup.inlineKeyboard(formattedSecondButtons));
 });
 
+bot.action('answering-first-order.4', async (ctx) => {
+  ctx.reply(secondPrompt, Markup.inlineKeyboard(formattedSecondButtons));
+});
 
 /*
-  Message received
+  COMPLETED ANSWERING
 */
-// https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression
-const emailRegex = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/
-bot.hears(emailRegex, async ctx => {
-  console.log("Got email")
-  const newUser: IBotUser = adaptCtx2User(ctx)
-  const {tg_id: tgId} = newUser
 
-  // check if user exists
-  const userExists = await doesUserExist(tgId)
+bot.action('answering-second-order.0', async (ctx) => {
+  const prompt =
+    'Well done! You just chomped a question.\n\nWhat do you want to do next?';
+  const buttonOptions: { [k: string]: string } = {
+    'completed-answering.more': 'Answer more 🎲',
+    'completed-answering.home': 'Go home 🏡',
+  };
+  const buttons = Object.keys(buttonOptions).map((key) =>
+    Markup.button.callback(buttonOptions[key], key),
+  );
+  ctx.reply(prompt, Markup.inlineKeyboard(buttons));
+});
 
-  if (userExists) {
-    ctx.reply("You already have an account. Please type /start to continue.")
-    return
-  } else {
-    const {match: emailAddresses} = ctx
-    const emailAddress = emailAddresses[0]
-    console.log(emailAddress)
-    newUser.original_email_address = emailAddress
-    newUser.state = EBotUserState.NEW
-
-    const headers = {
-      Authorization: `Bearer ${process.env.DYNAMIC_TOKEN}`,
-      'Content-Type': 'application/json'
-    }
+bot.action('answering-second-order.10', async (ctx) => {
+  const prompt =
```

Comment:
Would be nice to get rid of this code duplication and refactor logic into helper method.

Reply:
Yes! This part is temporary and it will be replaced with ideal composable code to handle the same requests on answer.